pkg/adapter/v2/cloudevents.go

```diff
@@ -111,6 +112,7 @@ func (c *client) reportCount(ctx context.Context, event cloudevents.Event, resul
 
 		_ = c.reporter.ReportEventCount(reportArgs, res.StatusCode)
 	} else {
+		crstatusevent.ReportCRStatusEvent(ctx, result)
```

Comment:
I wonder if it wouldn't be better to add a new client field for emitting events. That way we can provide different implementations, just in case decide to emit cloudevents instead of k8s events.

Reply:
Could want to do both?